### PR TITLE
Feature/dotnet6

### PR DESCRIPTION
**Pull request by sondreb**

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -51,6 +50,6 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <NoWarn>1591;1572;1573;1574</NoWarn>
     <DocumentationFile></DocumentationFile>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (dangershony) [inline src/Directory.Build.props:53]: Do we want his or is it a temporary change?

File: src/Features/Blockcore.Features.Consensus/Blockcore.Features.Consensus.csproj [hunk 2]
```diff
@@ -17,14 +17,17 @@
...
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.11" />
     <PackageReference Include="Microsoft.FASTER" Version="2020.2.1.1" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\Blockcore\Blockcore.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Microsoft.SourceLink.GitHub" Version="1.1.1" />
```
Comment (dangershony) [inline src/Features/Blockcore.Features.Consensus/Blockcore.Features.Consensus.csproj:28]: Why do we need a github package?

Comment (dangershony) [reply src/Features/Blockcore.Features.Consensus/Blockcore.Features.Consensus.csproj:28]: ok I see it's for creating debug info for debugging in to nuget packages

**Review by dangershony — APPROVED** [review]

**Comment by dangershony** [comment]
All unstable tests passed as well

File: src/Features/Blockcore.Features.Consensus/Blockcore.Features.Consensus.csproj (hunk 2 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.Consensus/Blockcore.Features.Consensus.csproj:28]: Yes, it's just an upgrade.

File: src/Directory.Build.props (hunk 1 above)
Comment (sondreb) [reply src/Directory.Build.props:53]: There are way too many fixes that must be done, so we'll leave it like this for now so it compiles.

Comment (dangershony) [reply src/Directory.Build.props:53]: ok fine so we can slowly fix warnings

**Comment by dangershony** [comment]
Push when ready

**Pull request merged**